src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -41,24 +41,24 @@ public override void Initialize(AnalysisContext context)
             // Don't report in generated code since that's not actionable.
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            context.RegisterCompilationStartAction(compilationContext =>
+            context.RegisterCompilationStartAction(context =>
             {
-                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationContext.Compilation);
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
 
                 // Get the list of all method' attributes for which the rule shall not be triggered.
                 ImmutableArray<INamedTypeSymbol> skippedAttributes = GetSkippedAttributes(wellKnownTypeProvider);
 
-                var isWebProject = compilationContext.Compilation.IsWebProject(compilationContext.Options);
+                var isWebProject = context.Compilation.IsWebProject(context.Options);
 
-                compilationContext.RegisterSymbolStartAction(
-                    symbolStartContext => OnSymbolStart(symbolStartContext, wellKnownTypeProvider, skippedAttributes, isWebProject),
+                context.RegisterSymbolStartAction(
+                    context => OnSymbolStart(context, wellKnownTypeProvider, skippedAttributes, isWebProject),
                     SymbolKind.NamedType);
             });
 
             return;
 
             static void OnSymbolStart(
-                SymbolStartAnalysisContext symbolStartContext,
+                SymbolStartAnalysisContext context,
```

Comment:
Is it necessary to rename those context parameters? I think the original ones were clearer

Reply:
➡️ We use `context` as the name of all context parameters to intentionally hide outer parameters with the same name, as they are not valid for use in the nested lambdas.